### DISCUSSION TITLE
Needs some step by step instructions on how to add multple domains.

### DISCUSSION THREAD

**Discussion by tonytonyistony** [modoboa/modoboa#2798] [Help] [answered]
I would appreciate some help from an vet.

I'm 3 days into messing with Modoboa and I'm liking it so far, however the multiple domains is getting me a little frustrated as there doesn't seem to be clear instruction on how to implement it.

https://www.linuxbabe.com/mail-server/modoboa-multiple-domains - I feel like this is what you shouldnt do given some of the responses from the dev in other posts.

This is what Ive done so far.
1. Created a second domain
2. Added it to the settings.py
3. Added the alias to the nginx config
4. Add DNS entries to the second domain as specified in the DNS popup in the GUI
5. Ive created a mailbox to test

I feel like its partially working.
Gmail can send and receive on this domain, but protonmail can receive but cant send to and says there is no mail domain and fails. I cannot connect a thunderbird client to the second domain via IMAP and SMTP.

I feel like Im missing configurations somewhere, but there is no documentation.

**Comment by Spitfireap** [comment, marked as answer]
Hi, 
I would say the right way is just use one domain to access your server (smtp, imap, http).

So for instance you create mail.domain1.tld, this is your base domain. When you add domain2.tld you still connect on mail.domain1.tld and when configuring a mail client you use mail.domain1.tld as an smtp and imap server.
Else you need to generate a new ssl cert for mail.domain2.tld, and add [this cert on dovecot configuration](https://serverfault.com/a/952994) (plus what you've already done)

**Reply by tonytonyistony** [reply]
> generate a new ssl cert for mail.domain2.tld, and add [this cert on dovecot configuration](https://serverfault.com/a/952994)
This is what I was missing

Could we get this info added to the "Read the Docs" site?

**Reply by tonytonyistony** [reply]
> So for instance you create mail.domain1.tld, this is your base domain. When you add domain2.tld you still connect on mail.domain1.tld and when configuring a mail client you use mail.domain1.tld as an smtp and imap server.
How does this effect uniqueness of the second domain? Like if you wanted to host another site for a friend, it doesnt make sense to give him my domain details for his domain. 

It sounds like if I want it to be completely unique I need to do the following.
- Create a second domain in the GUI
- Add your DNS settings as specified in the GUI
- Add domain to allowed list in settings.py
- Make a copy of the nginx site config and change the variables
- Issue SSL cert
- Add cert details to dovecot config

The last thing I would be curious about is rDNS. Can you have multiple rDNS entries on a single IP?
